File: 자료구조 구현/linkedList_stack.py
# # 11. 10828 연결리스트 스택
class Node:
    def __init__(self,value):
        self.value = value
        self.next = None

class Stack:
    def __init__(self):
        self.head = None
    
    def push(self,value):
        new_node = Node(value)
        new_node.next = self.head
        self.head = new_node
    def pop(self):
        tmp=self.head
        self.head = self.head.next
        return tmp.value

stack = Stack()
stack.push(1)
print(stack.pop())

File: 자료구조 구현/linkedList_stack.js
class Node {
  constructor(value) {
    this.value = value;
    this.next = null;
  }
}

class Stack {
  constructor() {
    this.top = null;
    this.size = 0;
  }

  push(value) {
    const newNode = new Node(value);
    newNode.next = this.top;
    this.top = newNode;
    this.size++;
  }

  pop() {
    if (this.top === null){
        return null;
    }
    let tmp = this.top.value;
    this.top = this.top.next;
    return tmp
  }
  
  contain(value){
    let currentNode = this.top;
    while (currentNode.next !== null){
      if(currentNode.value ===value){
        return true
      }else{
        currentNode = currentNode.next;
      }
    }
    return false
  }

}

let stack = new Stack();
stack.push(1);
stack.push(2);
stack.push(3);
console.log(stack.contain(3));
console.log(stack.contain(4));
console.log(stack.pop());
console.log(stack.pop());
